locator/abstract_replication_strategy.hh

```diff
@@ -206,6 +206,14 @@ public:
 
     virtual tablet_replica_set get_replicas_for_reading(const token& search_token) const = 0;
 
+    bool contains(const tablet_replica_set& replicas, inet_address host) const {
+        for (auto replica : replicas) {
+            if (this->get_token_metadata_ptr()->get_endpoint_for_host_id(replica.host) == host)
```

Comment:
We should work with host_id, not inet_address. Hosts can change IPs at unpredictable times.

The caller should use erm->get_token_metadata().get_my_id().